### PR TITLE
ENH: Create ArrowDtype

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Not activated yet. Opening to get any feedback about the approach cc @jbrockmendel

### CODE REVIEW

**Comment by jbrockmendel** [comment]
discussed on the March call, id like to migrate from BaseMaskedDtype/NumericDtype etc to just MaskedDtype.  Similarly id suggest here just ArrowDtype.

**Comment by mroeschke** [comment]
By "just ArrowDtype" you mean have `ArrowDtype` inherit directly from `ExtensionDtype`?

**Comment by jbrockmendel** [comment]
> By "just ArrowDtype" you mean have ArrowDtype inherit directly from ExtensionDtype?

Right, without all the subclasses

File: pandas/core/arrays/arrow/base.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+from pandas._typing import DtypeObj
+from pandas.util._decorators import cache_readonly
+
+from pandas.core.dtypes.base import StorageExtensionDtype
+
+from pandas.core.arrays.arrow import ArrowExtensionArray
+
+
+class ArrowDtype(StorageExtensionDtype):
```
Comment (jreback) [inline pandas/core/arrays/arrow/base.py:14]: why are the dtypes here? shouldn't these be in pandas/core/dtypes ?

Comment (mroeschke) [reply pandas/core/arrays/arrow/base.py:14]: I was thinking of consolidating all arrow related functionality under `/arrays/arrow/`. Thoughts @jbrockmendel

Comment (jbrockmendel) [reply pandas/core/arrays/arrow/base.py:14]: no objection.  Might be worth keeping _something_ in core.dtypes.dtypes for dependency-structure purposes

Comment (jreback) [reply pandas/core/arrays/arrow/base.py:14]: ok fair, looking to keep these separate as much as possible, maybe rename this to _dtype.py at some point.

**Comment by jreback** [comment]
> * [x]  All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
> 
> Not activated yet. Opening to get any feedback about the approach cc @jbrockmendel
> 
> 1. `BaseArrowDtype` inherits from `BaseMaskedDtype`
> 2. `NumericArrowDtype` mirrors `NumericDtype`

can we rename NumericDtype -> NumericMaskedDtype (alt maybe NumericNumpyDtype and rename BaseMaskedDtype)

**Comment by mroeschke** [comment]
> > * [x]   All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
> > 
> > Not activated yet. Opening to get any feedback about the approach cc @jbrockmendel
> > 
> > 1. `BaseArrowDtype` inherits from `BaseMaskedDtype`
> > 2. `NumericArrowDtype` mirrors `NumericDtype`
> 
> can we rename NumericDtype -> NumericMaskedDtype (alt maybe NumericNumpyDtype and rename BaseMaskedDtype)

This PR will just add the base ArrowDtype for now

**Pull request merged**